source/adios2/core/ADIOS.cpp

```diff
@@ -79,6 +79,18 @@ IO &ADIOS::DeclareIO(const std::string ioName)
     return ioPair.first->second;
 }
 
+IO &ADIOS::GetIO(const std::string name)
+{
+    auto itIO = m_IOs.find(name);
+    if (itIO == m_IOs.end())
```

Comment:
This one is a bit unclear.  If the only way to handle the name not found is to throw an exception since there's no way to return a "null" value for a reference object.  Give that the Declare method is also a reference object then I think it makes sense to keep this one as well.  If we would rather switch those to pointers instead of references, which I think is reasonable, so a null return is meaningful in the case of an error but no exception thrown, thhen I think it should be done in a seperate PR.